nutils/sample.py

```diff
@@ -336,9 +335,9 @@ def hull(self):
     return self.points.hull
 
   def get_evaluable_indices(self, ielem):
-    npoints = self.points.get_evaluable_coords(ielem).shape[0]
-    offsets = evaluable._SizesToOffsets(evaluable.loop_concatenate(evaluable.InsertAxis(npoints, 1), ielem, self.nelems))
```

Comment:
I'm in favor of keeping `_SizesToOffsets` here and removing `offsets`. With `getindex` using `get_evaluable_indices`, there's no reason to keep `offsets` around, which makes the entire `_DefaultIndex` cache-free. I'll admit that the offsets array is small compared to all other arrays we typically encounter during a simulation, but in some academic cases where for example the element matrix turns out to be constant, this might as well be one of the larger arrays.

`_SizesToOffsets`simplifies to a constant if the argument does not vary w.r.t. the loop index. The same happens with `Constant` of course, but that simplification is rather expensive compared with the former.

Reply:
And computing `_DefaultIndex.offsets` is done by looping over all elements. With `_SizesToOffsets` this loop is completely eliminated if the shapes do not vary.